### PR TITLE
[chore] Handle aliases in slices

### PR DESCRIPTION
**Pull request by cedric-cordenier**
## Updates

- Support unwrapping a list of bytes in values.Value to a list of alias types

```
// For example
type aliasUInt8 uint8

unwrapTo: []bytes to []aliasUInt8
```

### CODE REVIEW

File: pkg/values/value.go [hunk 1]
```diff
@@ -278,10 +278,36 @@ func unwrapTo[T any](underlying T, to any) error {
...
+			for i := 0; i < rUnderlying.Len(); i++ {
+				el := rUnderlying.Index(i)
+				toEl := reflect.New(rToVal.Type().Elem())
+				err := unwrapTo(el.Interface(), toEl.Interface())
+				if err != nil {
+					return err
+				}
+
+				if rToVal.Len() > i {
+					rToVal.Index(i).Set(reflect.Indirect(toEl))
```
Comment (kidambisrinivas) [inline pkg/values/value.go:301]: Does this require 'to' to be initialised with the right list size?

File: pkg/values/value.go [hunk 2]
```diff
@@ -278,10 +278,36 @@ func unwrapTo[T any](underlying T, to any) error {
...
+			reflect.Indirect(rTo.Convert(underlyingPtr)).Set(rUnderlying)
+			return nil
+		}
+
+		rToVal := reflect.Indirect(rTo)
+		if rToVal.Kind() == reflect.Slice && rUnderlying.Kind() == reflect.Slice {
+			for i := 0; i < rUnderlying.Len(); i++ {
+				el := rUnderlying.Index(i)
+				toEl := reflect.New(rToVal.Type().Elem())
+				err := unwrapTo(el.Interface(), toEl.Interface())
```
Comment (kidambisrinivas) [inline pkg/values/value.go:295]: Aha! Nice trick to recursively call unwrapTo and handle. In the base case where unwrapTo is called for each element, how is the following call to Set not throwing the same error:

```
reflect.Indirect(rTo.Convert(underlyingPtr)).Set(rUnderlying)
```

Error: 
```
panic: reflect: reflect.Value.Set using unaddressable value [recovered]
panic: reflect: reflect.Value.Set using unaddressable value
```

File: pkg/values/value_test.go [hunk 3]
```diff
@@ -382,3 +382,71 @@ func Test_Copy(t *testing.T) {
...
+				}
+
+				return to
+			},
+		},
+		{
+			name:    "aliasMap -> map[string]any",
+			val:     func() any { return map[string]any{} },
+			alias:   func() any { return aliasMap{} },
+			convert: func(a any) any { return map[string]any(a.(aliasMap)) },
```
Comment (kidambisrinivas) [inline pkg/values/value_test.go:431]: Nice, so the unwrap method works for aliases of complex types like map[string]any as well

**Review by kidambisrinivas — APPROVED** [review]
LGTM

File: pkg/values/value.go (hunk 2 above)
Comment (cedric-cordenier) [reply pkg/values/value.go:295]: Hey! Yeah, we avoid that problem by creating the pointer on line 294

Comment (kidambisrinivas) [reply pkg/values/value.go:295]: Ah ok. So thats what we were missing yesterday! Thanks!

**Review by kidambisrinivas — DISMISSED** [review]

**Review by kidambisrinivas — APPROVED** [review]

**Review by agparadiso — APPROVED** [review]

**Pull request merged**